drgn/helpers/linux/pid.py

```diff
@@ -123,3 +125,39 @@ def for_each_task_in_group(
     ):
         if other != task or include_self:
             yield other
+
+
+def is_kthread(t: Object) -> bool:
+    """
+    Check if a task is kernel thread.
+    """
+    if not t.mm and not task_state_to_char(t) == "Z":
+        return True
+    return False
+
+
+def is_user(t: Object) -> bool:
+    """
+    Check if a task is user thread.
+    """
+    if t.mm:
+        return True
+    elif task_state_to_char(t) == "Z":
+        return True
+    return False
+
+
+def is_group_leader(t: Object) -> bool:
+    """
+    Check if a task is thread group leader.
+    """
+    if t.pid.value_() == t.tgid.value_():
+        return True
+    return False
```

Comment:
The equivalent in the function in the kernel is [`thread_group_leader()`](https://github.com/torvalds/linux/blob/347e9f5043c89695b01e66b3ed111755afcf1911/include/linux/sched/signal.h#L700-L703), which has a drastically different implementation:

```c
static inline bool thread_group_leader(struct task_struct *p)
{
	return p->exit_signal >= 0;
}
```

I think we should base our helper on that, and put it in `drgn.helpers.linux.sched`.